test/index.js

```diff
@@ -0,0 +1,211 @@
+    var info = gpi('dimensions.lengths[2][1]', obj);
+    assert(info.parent === obj.dimensions.lengths[2]);
+    assert(info.value === obj.dimensions.lengths[2][1]);
+    assert(info.name === 1);
+    assert(info.exists === true);
+  });
+
+  it('should handle out of bounds array index', function () {
+    var info = gpi('dimensions.lengths[3]', obj);
+    assert(info.parent === obj.dimensions.lengths);
+    assert(info.value === undefined);
+    assert(info.name === 3);
+    assert(info.exists === false);
+  });
+
+  it('should handle out of bounds dimensional array index', function () {
+    var info = gpi('dimensions.lengths[2][5]', obj);
+    assert(info.parent === obj.dimensions.lengths[2]);
+    assert(info.value === undefined);
+    assert(info.name === 5);
+    assert(info.exists === false);
+  });
+
+  it('should handle backslash-escaping for .[]', function () {
+    var info = gpi('dimensions\\.lengths.\\[2\\][1]', obj);
+    assert(info.parent === obj['dimensions.lengths']['[2]']);
+    assert(info.value === obj['dimensions.lengths']['[2]'][1]);
+    assert(info.name === 1);
+    assert(info.exists === true);
+  });
+});
+
+describe('getPathValue', function () {
+  it('returns the correct value', function () {
+    var object = {
+      hello: 'universe',
+      universe: {
+        hello: 'world',
+      },
+      world: [ 'hello', 'universe' ],
+      complex: [
+        { hello: 'universe' },
+        { universe: 'world' },
+        [ { hello: 'world' } ],
+      ],
+    };
+
+    var arr = [ [ true ] ];
+    assert(pathval.getPathValue('hello', object) === 'universe');
+    assert(pathval.getPathValue('universe.hello', object) === 'world');
+    assert(pathval.getPathValue('world[1]', object) === 'universe');
+    assert(pathval.getPathValue('complex[1].universe', object) === 'world');
+    assert(pathval.getPathValue('complex[2][0].hello', object) === 'world');
+    assert(pathval.getPathValue('[0][0]', arr) === true);
+  });
+
+  it('handles undefined objects and properties', function () {
+    var object = {};
+    assert(pathval.getPathValue('this.should.work', undefined) === null);
+    assert(pathval.getPathValue('this.should.work', object) === null);
+    assert(pathval.getPathValue('length', 'word') === 4);
+  });
+});
+
+describe('setPathValue', function () {
+  it('allows value to be set in simple object', function () {
+    var obj = {};
+    pathval.setPathValue('hello', obj, 'universe');
+    assert(obj.hello === 'universe');
+  });
+
+  it('allows nested object value to be set', function () {
+    var obj = {};
+    pathval.setPathValue('hello.universe', obj, 'properties');
+    assert(obj.hello.universe === 'properties');
+  });
+
+  it('allows nested array value to be set', function () {
+    var obj = {};
+    pathval.setPathValue('hello.universe[1].properties', obj, 'galaxy');
+    assert(obj.hello.universe[1].properties === 'galaxy');
+  });
+
+  it('allows value to be REset in simple object', function () {
```

Comment:
Yes, it was already like that before this change so I thought it would be better to leave it like this.
It aims at emphasizing we are setting a value again (`RE`setting).